lib/dogma/rules/hard_tabs.ex

```diff
@@ -0,0 +1,35 @@
+defmodule Dogma.Rules.HardTabs do
+  @moduledoc """
+  Requires that all indentation is done using spaces rather than hard tabs.
+
+  So the following would be invalid:
+
+  def something do
+  \t:body
+  end
+  """
+  @behaviour Dogma.Rule
+
+  @indenting_tab_pattern ~r/^\t+.+/
```

Comment:
Well I can't think of one (but then I don't really use hard tabs). Mainly just done it like this because the thing we want to stop is hard tabs being used to control indenting. Rather than tabs existing in the file at all.